### PR TITLE
Various refactoring to improve code readability

### PR DESCRIPTION
**Pull request by alexjfisher**
Refactor (and hopefully simplify) some of the conditional logic and relationships.

I've initially submitted this as a series of commits to make it easier to review (and find mistakes!).
If everything looks ok, I'll rebase and squash before merging.

In a future PR it should then be easy to move resources to separate classes and follow the `Install -> Config ~> Service` pattern.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -437,6 +432,16 @@
     }
   }
 
+  if $::osfamily == 'Suse' {
+    exec { 'install /etc/init.d/snmptrapd':
```
Comment (ekohl) [inline manifests/init.pp]: Equivalent:
```puppet
file { '/etc/init.d/snmptrapd':
  source  => '/usr/share/doc/packages/net-snmp/rc.snmptrapd',
  owner   => 'root',
  group   => 'root',
  mode    => '0755',
  require => Package['snmpd'],
  before  => Service['snmptrapd'],
}
```
Possibly with a `replace => false` to emulate the `creates =>` on `exec`.

File: spec/classes/snmp_client_spec.rb [hunk 2]
```diff
@@ -51,9 +51,8 @@
             mode: '0644',
             owner: 'root',
             group: 'root',
-            path: '/etc/snmp/snmp.conf',
-            require: ['Package[snmp-client]', 'File[/etc/snmp]']
-          )
+            path: '/etc/snmp/snmp.conf'
```
Comment (ekohl) [inline spec/classes/snmp_client_spec.rb]: I'd be happy with trailing commas. Much smaller diff :)

Comment (alexjfisher) [reply spec/classes/snmp_client_spec.rb]: rubocop won't be though :(

Comment (ekohl) [reply spec/classes/snmp_client_spec.rb]: We may have to teach rubocop a lesson then. There's an option for it.

Comment (alexjfisher) [reply spec/classes/snmp_client_spec.rb]: meh.  Just think of the noise correcting all 100 odd modules to the new style! ;)

File: manifests/init.pp (hunk 1 above)
Comment (alexjfisher) [reply manifests/init.pp]: PR updated.  I thought about the `replace => false` and decided it probably isn't needed.

File: spec/classes/snmp_client_spec.rb (hunk 2 above)
Comment (vStone) [reply spec/classes/snmp_client_spec.rb]: thats what the rubocop:autocorrect task is for? :)

**Comment by alexjfisher** [comment]
@ekohl @vStone This is about as much refactoring I want to put in a single PR.  If you're happy that it's correct (and an improvement!), could you approve it?  I'll then rebase/squash, fix up the PR title etc and merge.

**Comment by alexjfisher** [comment]
Cool.  I'll rebase now.

**Comment by alexjfisher** [comment]
Of course, it always helps to do a `git pull` before rebasing and force pushing...  (I accidentally threw away the last commit that fixed up the tests because I did it on a different workstation).

**Pull request merged**